packages/medusa/src/api/routes/admin/products/create-product.ts

```diff
@@ -1,10 +1,29 @@
-import { MedusaError, Validator } from "medusa-core-utils"
-import { defaultRelations, defaultFields } from "."
+import { Type } from "class-transformer"
+import {
+  IsArray,
+  IsBoolean,
+  IsEnum,
+  IsInt,
+  IsNumber,
+  IsObject,
+  IsOptional,
+  IsString,
+  Validate,
+  ValidateNested,
+} from "class-validator"
+import {
+  defaultAdminProductFields,
+  defaultAdminProductRelations,
+  ProductStatus,
+} from "."
+import { XorConstraint } from "../../../../types/validators/xor"
+import { validator } from "../../../../utils/validator"
 
 /**
  * @oas [post] /products
  * operationId: "PostProducts"
  * summary: "Create a Product"
+ * x-authenticated: true
  * description: "Creates a Product"
  * requestBody:
```

Comment:
Should we add required fields for at least the top level properties of the product